BootloaderCommonPkg/Library/BootloaderCommonLib/BootloaderCommonLib.c

```diff
@@ -297,87 +297,115 @@ DetectUsedStackBottom (
-  OUT UINT32     *Size
+GetComponentEntryByPartition (
+  IN  UINT32                Signature,
+  IN  BOOLEAN               IsBackupPartition
   )
 {
   UINTN                 Index;
   UINT32                MaxEntries;
-  UINT32                PcdBase;
-  UINT32                PcdSize;
-  UINT32                RomBase;
   FLASH_MAP             *FlashMapPtr;
-  EFI_STATUS            Status;
-  FLASH_MAP_ENTRY_DESC  EntryDesc;
-
-  PcdBase = 0;
-  PcdSize = 0;
-  Status = EFI_NOT_FOUND;
+  FLASH_MAP_ENTRY_DESC  *EntryDesc;
 
   FlashMapPtr = GetFlashMapPtr ();
   if (FlashMapPtr == NULL) {
-    return EFI_UNSUPPORTED;
+    return NULL;
   }
 
-  RomBase = (UINT32) (0x100000000ULL - FlashMapPtr->RomSize);
   MaxEntries = ((FlashMapPtr->Length - FLASH_MAP_HEADER_SIZE) / sizeof (FLASH_MAP_ENTRY_DESC));
 
   for (Index = 0; Index < MaxEntries; Index++) {
-    EntryDesc = FlashMapPtr->EntryDesc[Index];
+    EntryDesc = (FLASH_MAP_ENTRY_DESC *)&FlashMapPtr->EntryDesc[Index];
     //
     // Look for the component with desired signature
     //
-    if (EntryDesc.Signature == 0xFFFFFFFF) {
-      Status = EFI_NOT_FOUND;
+    if (EntryDesc->Signature == 0xFFFFFFFF) {
       break;
     }
-    if (EntryDesc.Signature == Signature) {
+    if (EntryDesc->Signature == Signature) {
       //
       // Check if need to get back up copy
       // Back up copies can be identified with back up flag
       //
-      if ( ((EntryDesc.Flags & (FLASH_MAP_FLAGS_NON_REDUNDANT_REGION | FLASH_MAP_FLAGS_NON_VOLATILE_REGION)) != 0) ||
-           (((IsBackupPartition ? FLASH_MAP_FLAGS_BACKUP : 0) ^ (EntryDesc.Flags & FLASH_MAP_FLAGS_BACKUP)) == 0) ) {
-        PcdBase = (UINT32) (RomBase + EntryDesc.Offset);
-        PcdSize = EntryDesc.Size;
-        Status = EFI_SUCCESS;
-        break;
+      if ( ((EntryDesc->Flags & (FLASH_MAP_FLAGS_NON_REDUNDANT_REGION | FLASH_MAP_FLAGS_NON_VOLATILE_REGION)) != 0) ||
+           (((IsBackupPartition ? FLASH_MAP_FLAGS_BACKUP : 0) ^ (EntryDesc->Flags & FLASH_MAP_FLAGS_BACKUP)) == 0) ) {
+        return EntryDesc;
       }
     }
   }
 
+  return NULL;
+}
+
+/**
+  Gets component information from the flash map by partition.
+
+  This function will look for the component matching the input signature
+  in the flash map, if found, it will look for the component with back up
+  flag based on the backup partition parmeter and will return the
+  base address and size of the component.
+
+  @param[in]  Signature         Signature of the component information required
+  @param[in]  IsBackupPartition TRUE for Back up copy, FALSE for primary copy
+  @param[out] Base              Base address of the component
+  @param[out] Size              Size of the component
+
+  @retval    EFI_SUCCESS    Found the component with the matching signature.
+  @retval    EFI_NOT_FOUND  Component with the matching signature not found.
+
+**/
+EFI_STATUS
+EFIAPI
+GetComponentInfoByPartition (
+  IN  UINT32     Signature,
+  IN  BOOLEAN    IsBackupPartition,

```

Comment:
IsBackupPartition is not used in this function, is it expected?

Reply:
Fixed